### PR TITLE
Added twilights to the nautical navigation data

### LINKED ISSUE #408: Nautical Navigation Data
_Original report:_ https://bugs.launchpad.net/stellarium/+bug/1106745

This feature request from old bug tracker.

Object Data
 - [x] Rise, Meridian Passage and Set times
 - [x] GHA (Greenwich Hour Angle)
 - [x] SHA (Sidereal Hour Angle)
 - [x] LHA (Local Hour Angle)

Twilights (start & end times)
 - [x] civil
 - [ ] nautical
 - [ ] astronomical

_An additional proposal from the old bug tracker:_ indicate beginning and ending times of all twilights for the selected location in the sun data... also moon rise and set data.

### PR DESCRIPTION
**Pull request by alex-w**
### Description
This PR contains the new tab in Navigational Stars plugin with twilights data. See issue 408 to details.

Fixes #408 (issue)

### Screenshot
stellarium-005

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by alex-w** [comment]
@gzotti should I add noon and midnight times and duration of day/night in addition to the twilight data?

File: plugins/NavStars/src/gui/NavStarsWindow.cpp [hunk 1]
```diff
@@ -106,6 +111,87 @@ void NavStarsWindow::resetSettings()
...
+	ui->labelNauticalTwilightBegin->setText(nauticalTwilightBegin);
+	ui->labelNauticalTwilightEnd->setText(nauticalTwilightEnd);
+	ui->labelAstronomicalTwilightBegin->setText(astronomicalTwilightBegin);
+	ui->labelAstronomicalTwilightEnd->setText(astronomicalTwilightEnd);
+
+	// tooltips
+	// TRANSLATORS: full phrase is "XX° below the horizon"
+	QString belowHorizon = q_("° below the horizon");
+	// TRANSLATORS: full phrase is "XX° above the horizon"
+	QString aboveHorizon = q_("° above the horizon");
```
[the remark this answers is not in the document]
Comment (alex-w) [reply plugins/NavStars/src/gui/NavStarsWindow.cpp:187]: Yes, of course

**Comment by gzotti** [comment]
I have not tested this yet. Do you add extraInfoStrings? Probably allow selection of which data the users wants to see, but it may be a meaningful addition, yes. Also moonrise/-set, if not yet included.  Sorry, I have no time likely until next Wednesday.

**Comment by alex-w** [comment]
> Do you add extraInfoStrings? Probably allow selection of which data the users wants to see, but it may be a meaningful addition, yes. 

No. This is the tab with extra data ("Twilights") within the plugin. Info strings for the Sun already have the configurable twilight data.

> Also moonrise/-set, if not yet included. 

Hmm... It can be useful for navigation.

**Comment by alex-w** [comment]
Moonrise/moonset and day duration info was added

**Comment by alex-w** [comment]
stellarium-006

**Comment by alex-w** [comment]
OK, I think it's ready to merge now

File: src/gui/AstroCalcDialog.cpp [hunk 2]
```diff
@@ -7207,7 +7207,7 @@ void AstroCalcDialog::calculateWutObjects()
 		ui->wutAngularSizeLimitMaxSpinBox->setToolTip(q_("Maximum angular size for visible celestial objects"));
 
 		// Direct calculate sunrise/sunset (civil twilight)
-		Vec4d rts = GETSTELMODULE(SolarSystem)->getSun()->getRTSTime(core, -7.);
+		Vec4d rts = GETSTELMODULE(SolarSystem)->getSun()->getRTSTime(core, -6.);
```
Comment (gzotti) [inline src/gui/AstroCalcDialog.cpp:7210]: I would call this differently, e.g. rtsM6 or rtsCivil, to make the code clearer.

Comment (alex-w) [reply src/gui/AstroCalcDialog.cpp:7210]: This variable re-used in other parts of this method

**Comment by alex-w** [comment]
@gzotti please check the changes

File: src/gui/AstroCalcDialog.cpp (hunk 2 above)
Comment (gzotti) [reply src/gui/AstroCalcDialog.cpp:7210]: Ah, OK.

**Review by gzotti — APPROVED** [review]
Fine for me!

**Pull request merged**